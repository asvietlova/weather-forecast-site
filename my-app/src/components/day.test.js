import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Day from "./day";

Enzyme.configure({ adapter: new Adapter() });

describe("Day", () => {
  let wrapper;

  beforeEach(() => {
    let dayInfo = {
      id: 7,
      date: "27/02/2022",
      temperatureMin: "-2",
      temperatureMax: "+3",
      atmPressure: "750",
      humidity: "62",
      wind: "2.0",
      description: "снег",
      img_url: "./img/snow.png",
    };

    wrapper = shallow(<Day {...dayInfo} />);
  });

  test("should contain class day", () => {
    expect(wrapper.find(".day").exists()).toBe(true);
    expect(wrapper.find(".day").length).toBe(1);
  });

  test("should contain img", () => {
    expect(wrapper.find("img").exists()).toBe(true);
    expect(wrapper.find("img").length).toBe(1);
  });

  test("should match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
