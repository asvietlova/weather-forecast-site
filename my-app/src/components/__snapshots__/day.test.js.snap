// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day should match snapshot 1`] = `"<div class=\\"day\\"><p>воскресенье, 27 февраля 2022 г.</p><img src=\\"./img/snow.png\\" alt=\\"снег\\"/><div class=\\"temperature\\"><div><p>мин.</p><p>-2°</p></div><div><p>макс.</p><p>+3°</p></div></div><p>Давление, мм: <span>750</span></p><p>Влажность, %: <span>62</span></p><p>Ветер, м/сек.: <span>2.0</span></p></div>"`;
